flytepropeller/pkg/controller/nodes/attr_path_resolver.go

```diff
@@ -46,8 +46,7 @@ func resolveAttrPathInPromise(ctx context.Context, datastore *storage.DataStore,
 			}
 			currVal = currVal.GetCollection().GetLiterals()[attr.GetIntValue()]
 			index++
-		// scalar is always the leaf, so we can break here
-		case *core.Literal_Scalar:
+		default:
 			break
```

Comment:
this is the right logic, we should stop the iteration if this is not a map, collection, or offloaded literal.